app/views/spree/admin/variants/_form.html.haml

```diff
@@ -64,41 +65,3 @@
       = f.text_field field, value: value, class: 'fullwidth'
 
 .clear
-
-
-:javascript
-  angular.element(document.getElementById("new_variant")).ready(function() {
-    angular.bootstrap(document.getElementById("new_variant"), ['admin.products']);
-  });
-
-  $(document).ready(function() {
-
-      var on_demand = $('input#variant_on_demand');
-      var on_hand = $('input#variant_on_hand');
-
-      disableOnHandIfOnDemand = function() {
-          on_demand_checked = on_demand.attr('checked')
-          if ( on_demand_checked == undefined )
-              on_demand_checked = false;
```

Comment:
have you verified this is not required? does the new code work when on_demand is null?